fix(bredcrumbs): synced with new design changes

agliga opened the pull request:
Fixes #1918

- [X] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
* Basically spoke to design and got some new requirements. These should address that.
* Breadcrumbs are a max-width of 168px. They will overflow now.
* The min breadcrumbs is 2 + overflow (root, overflow, currentpage), and max is 4 (either 4 breadcrumbs, or root, overflow, patent page, current page). Updated the docs for that.

## Notes
One of the issues I ran into on the ebayui side was that the breadcrumbs were using menu, but instead we should be using `fake-menu` which automatically supports links/buttons. I updated the code to take that.
There was some issues some of the styles bleeding into the menu. I updated all the breadcrumb selectors for `ul`, `li` to be immediate children. Also removed the nested less structure, since Ian likes it more that way


- [X] I verify the build is in a non-broken state
- [X] I verify all changes are within scope of the linked issue
- [X] I regenerated all CSS files under dist folder
- [X] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [X] I tested the UI in dark mode and RTL mode
- [X] I added/updated/removed Storybook coverage as appropriate

ArtBlue reviewed (approved):
Looks good. I modified some stuff in the docs, adjusted some things to force the ellipses, and added the min states in stories.

agliga commented:
> Looks good. I modified some stuff in the docs, adjusted some things to force the ellipses, and added the min states in stories.

Great. Changes lgtm overall
